owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerConnector.java

```diff
@@ -71,7 +75,11 @@ protected final void handleMessage(PeerMessage message) {
       case REQUEST:
         break;
       case PIECE:
-        storageAdapter.write(((PieceMessage) message).getFilePiece());
+        FileBlock fileBlock = ((PieceMessage) message).getFileBlock();
+        if (manager.reportAndValidatePieceInProgress(fileBlock)) {
+          storageAdapter.write(peer.getTorrent(), fileBlock);
```

Comment:
if this throws an error we need to tell the manager we failed and release the hold on the block. I can do in my pr, because right now the adapter doesn't throw errros

Reply:
Ok, thanks!